test/spec/upload.js

```diff
@@ -169,50 +184,55 @@ describe("tus", function () {
       done();
     });
 
-    it("should resolve relative URLs", function () {
+    it("should resolve relative URLs", function (done) {
       var file = new Blob("hello world".split(""));
+      var host = "http://relative.tus.io";
       var options = {
-        endpoint: "http://master.tus.io:1080/files/"
+        endpoint: host + "/files/"
       };
 
       var upload = new tus.Upload(file, options);
       upload.start();
 
-      var req = jasmine.Ajax.requests.mostRecent();
-      expect(req.url).toBe("http://master.tus.io:1080/files/");
-      expect(req.method).toBe("POST");
+      waitTillNextReq(host, null, function (req) {
+        expect(req.url).toBe(host + "/files/");
+        expect(req.method).toBe("POST");
 
-      req.respondWith({
-        status: 201,
-        responseHeaders: {
-          "Location": "//localhost/uploads/foo"
-        }
-      });
+        req.respondWith({
+          status: 201,
+          responseHeaders: {
+            "Location": "//relative.tus.io/uploads/foo"
```

Comment:
By changing the Location header you removed on of the purposes of this test. Please change it back, so that the hostname from the Location header is different than the one from the endpoint.

Reply:
Addressed in https://github.com/tus/tus-js-client/pull/152.